### PR TITLE
Remove references to non-existent Diversity Advisory Council 

### PR DESCRIPTION
**Pull request by Erin007**
While investigating an accessibility bug [A11Y-10 ](https://codedotorg.atlassian.net/browse/A11Y-10) on https://code.org/about/diversity-council I discovered that page 404s because we no longer have a Diversity Advisory Council! I confirmed this with Katie Hendrickson.

Screenshot 2023-12-11 at 2 08 51 PM

 As clean up, this PR removes the 2 remaining references to the "Diversity Advisory Council" on the site: 

1.) updates the [Donation Policy](https://code.org/about/donation-policy) to read "Leadership Team, our Board of directors, and select advisors" rather than "[Leadership Team, our Board of directors, select advisors, and (in relevant cases) our Diversity Advisory Council". 

2.) removes the un-used alt-about-nav view which references the non-existent Advisory Council 

## Warning!!

We have entered Pixel Lock for Hour of Code!

Computer Science Education Week will be happening from Dec 4 - Dec 10. Alongside this event, we will
be launching our new Hour of Code activity. Please consider any risk introduced in this PR that
could affect Dance Lab, instructions, saving and logging student progress, caching, or anything
related to the Hour of Code activities, old or new. Even small changes, such as a different button
color, are considered significant during this time. If this change will affect the new Hour of Code
activity in any way, join the morning change review to get your changes approved prior to merging.
Reach out to the Student Labs team for more details!

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: pegasus/sites.v3/code.org/views/alt-about_nav.haml [hunk 1]
```diff
@@ -1,11 +0,0 @@
-#about_nav
```
Comment (hannahbergam) [inline pegasus/sites.v3/code.org/views/alt-about_nav.haml:1]: Woah so this page wasn't being touched at all???

Comment (Erin007) [reply pegasus/sites.v3/code.org/views/alt-about_nav.haml:1]: I can't find anywhere we render this view and it came in as part of staging content scoop changes 3 years ago. Not only does this view reference the non-existent Diversity Council, it also links to code.org/about/alt-about/our-impact which isn't anything (as far as I can tell) either. 🤷‍♀️  I'm fairly confident it's just leftover from some long ago header work, but is there anyone else we should confirm with?

Comment (kelbyhawn) [reply pegasus/sites.v3/code.org/views/alt-about_nav.haml:1]: We can definitely get rid of this! 😄

**Review by kelbyhawn — APPROVED** [review]
Looks good!

**Review by hannahbergam — APPROVED** [review]
Seeing @kelbyhawn 's comments, seems right to me! Thanks Erin! Also, I'm curious about the disbanding of the diversity council but that's not a discussion for here

**Pull request merged**